### DISCUSSION TITLE
Which bert training program should I use?

### DISCUSSION THREAD

**Discussion by timercrack** [Cryolite/kanachan#46] [Q&A] [answered]
There are three bert trainning program called phase1, phase2 and phase 3(on develop branch), are there any difference? which one should I use?

**Comment by shinkuan** [comment]
The goal of each is different. For example: Phase1's goal is to imitates human choices.

**Reply by timercrack** [reply]
Thank you for your reply. From the source code I saw phase2 is exactly the same as phase1 except it did some extra processing on the training data. What is the effect on the training results? Do I need to do the same thing to my data set?

**Reply by Cryolite** [reply, marked as answer]
In phase1, the model learns to imitate human behavior in the training data. In phase2, it learns to predict the point delta of each round based on the choices made in the training data. The training data contains both human actions and the point delta of each round, and the only difference between phase1 and phase2 is what they calculate as their objective function.

**Reply by timercrack** [reply]
Which one should I choose if the encoder is used for iql training later?

**Reply by Cryolite** [reply]
I have never actually tried any experiment of transferring the encoder trained with kanachan.training.bert.phase[12] to IQL, so I cannot say for sure. However, it is experimentally known that training with phase1 yields a better model, so there may be a higher likelihood that transferring the encoder trained in phase1 would be more beneficial.

**Reply by timercrack** [reply]
ok, when I finished training I will post the results here.
